src/Bex/Behat/ScreenshotExtension/Service/ScreenshotUploader.php

```diff
@@ -21,20 +20,29 @@ class ScreenshotUploader
     private $config;
 
     /**
-     * @var EventDispatcherInterface
+     * @var array
+     *   Used to statically store the list of files uploaded.
      */
-    private $eventDispatcher;
+    private static $imageUrls;
```

Comment:
Why is it static? This service will be kept in the service container all the time, so this instance will be available during test execution.

Reply:
Check that

With static it kind of works, but it wrong, because we can see attachments from other scenarios, so need to fix that.
<img width="994" alt="Screenshot 2023-01-19 at 17 17 50" src="https://user-images.githubusercontent.com/2162363/213515083-8cb629b2-f75f-4a0e-8778-81d926f77064.png">


I tried without using static, when the Cucumber tries to get the values, it comes as NULL. Not sure if I am doing something wrong. Need to investigate further...
<img width="1054" alt="Screenshot 2023-01-19 at 17 14 50" src="https://user-images.githubusercontent.com/2162363/213514664-cb7c010a-bfa7-4487-b429-510f2b6b16c1.png">

Reply:
Me neither, but using `static` in an OOP language is a code smell. 
Maybe you just need to subscribe to a different event and/or switch before/after.
Unfortunately I can't give you a better idea without digging down in the Behat source code.